wd_parallel_pkg/wd_parallel/collectives.py

```
@@ -0,0 +1,618 @@
+
+        if seq_dim == 0:
+            out_shape = (input_.size(0) * world_size,) + input_.shape[1:]
+            output = torch.empty(out_shape, dtype=input_.dtype, device=input_.device)
+            work = _launch_collective_async(
+                input_,
+                lambda: dist.all_gather_into_tensor(output, input_.contiguous(), group=group, async_op=True),
+            )
+            result = output
+        else:
+            x0 = input_.transpose(0, seq_dim).contiguous()
+            out_shape = (x0.size(0) * world_size,) + x0.shape[1:]
```

Ruff (0.15.11)

[warning] Consider `(x0.size(0) * world_size, *x0.shape[1:])` instead of concatenation

Replace with `(x0.size(0) * world_size, *x0.shape[1:])`

(RUF005)